### PR TITLE
[staging] openssl: fix bin installation for static builds

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
Fix bin dir installation for static builds

#119398
before:
```
$ tree /nix/store/zbr5skxl8j7y4xyqr7vip3hj75g5g41m-openssl-1.0.2e-bin/
/nix/store/zbr5skxl8j7y4xyqr7vip3hj75g5g41m-openssl-1.0.2e-bin/
├── c_rehash
└── openssl
```
after
```
$ tree result-bin
result-bin
└── bin
    ├── c_rehash
    └── openssl
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lukegb** [comment]
Hah, whoops. LGTM.

**Review by lukegb — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119825` at c6469b95 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1358 packages marked as broken and skipped:</summary>
  <ul>
    <li>Literate</li>
    <li>OVMF-CSM</li>
    <li>acl2</li>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>agdaPackages.iowa-stdlib</li>
    <li>aldor</li>
    <li>alliance</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>23883 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>CHOWTapeModel</li>
    <li>DisnixWebService</li>
    <li>EBTKS</li>
    <li>EmptyEpsilon</li>
    <li>Fabric (python38Packages.Fabric)</li>
    <li>LAStools</li>
    <li>LASzip</li>
    <li>LASzip2</li>
    <li>MACS2</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>254 packages built successfully:</summary>
  <ul>
    <li>asciidoc</li>
    <li>cacert</li>
    <li>ddgr</li>
    <li>docutils (python38Packages.docutils)</li>
    <li>fribidi</li>
    <li>giflib</li>
    <li>gnu-efi</li>
    <li>python3 (gnuradioPackages.python ,python38 ,python38Packages.python ,sourcehut.python)</li>
    <li>itstool</li>
    <li>jpylyzer (python38Packages.jpylyzer)</li>
    <li>kmod</li>
    <li>libapparmor</li>
    <li>libevdev</li>
    <li>libglvnd</li>
    <li>libssh2</li>
    <li>libsysprof-capture</li>
    <li>libvdpau</li>
    <li>meson</li>
    <li>mpvScripts.sponsorblock</li>
    <li>mpvScripts.thumbnail</li>
    <li>ninja</li>
    <li>openssl (openssl_1_1)</li>
    <li>pciutils</li>
    <li>postiats-utilities</li>
    <li>python38Packages.Babel</li>
    <li>python38Packages.alabaster</li>
    <li>python38Packages.apipkg</li>
    <li>python38Packages.appdirs</li>
    <li>python38Packages.async-timeout</li>
    <li>python38Packages.async_generator</li>
    <li>python38Packages.atomicwrites</li>
    <li>python38Packages.attrs</li>
    <li>python38Packages.backports-entry-points-selectable</li>
    <li>python38Packages.brotli</li>
    <li>python38Packages.certifi</li>
    <li>python38Packages.click</li>
    <li>python38Packages.cloudpickle</li>
    <li>python38Packages.colorama</li>
    <li>python38Packages.constantly</li>
    <li>python38Packages.coverage</li>
    <li>python38Packages.cppy</li>
    <li>python38Packages.cycler</li>
    <li>python38Packages.dateutil (python38Packages.python-dateutil)</li>
    <li>python38Packages.distlib</li>
    <li>python38Packages.django (python38Packages.django_2)</li>
    <li>python38Packages.eradicate</li>
    <li>python38Packages.execnet</li>
    <li>python38Packages.extras</li>
    <li>python38Packages.filelock</li>
    <li>python38Packages.flake8</li>
    <li>python38Packages.flaky</li>
    <li>python38Packages.flit-core</li>
    <li>python38Packages.freezegun</li>
    <li>python38Packages.gunicorn</li>
    <li>python38Packages.hyperlink</li>
    <li>python38Packages.idna</li>
    <li>python38Packages.imagesize</li>
    <li>python38Packages.importlib-metadata</li>
    <li>python38Packages.incremental</li>
    <li>python38Packages.iniconfig</li>
    <li>python38Packages.intreehooks</li>
    <li>python38Packages.iso8601</li>
    <li>python38Packages.itsdangerous</li>
    <li>python38Packages.jinja2</li>
    <li>python38Packages.lazy-object-proxy</li>
    <li>python38Packages.libevdev</li>
    <li>python38Packages.libselinux</li>
    <li>python38Packages.libxml2</li>
    <li>python38Packages.libxslt</li>
    <li>python38Packages.linecache2</li>
    <li>python38Packages.m2r</li>
    <li>python38Packages.markupsafe</li>
    <li>python38Packages.mccabe</li>
    <li>python38Packages.mistune (python38Packages.mistune_0_8)</li>
    <li>python38Packages.mock</li>
    <li>python38Packages.more-itertools</li>
    <li>python38Packages.nose</li>
    <li>python38Packages.nose-exclude</li>
    <li>python38Packages.nosexcover</li>
    <li>python38Packages.olefile</li>
    <li>python38Packages.outcome</li>
    <li>python38Packages.packaging</li>
    <li>python38Packages.parameterized</li>
    <li>python38Packages.pbr</li>
    <li>python38Packages.pexpect</li>
    <li>python38Packages.pip</li>
    <li>python38Packages.pkgconfig</li>
    <li>python38Packages.pluggy</li>
    <li>python38Packages.pretend</li>
    <li>python38Packages.psutil</li>
    <li>python38Packages.ptyprocess</li>
    <li>python38Packages.py</li>
    <li>python38Packages.pyasn1</li>
    <li>python38Packages.pycodestyle</li>
    <li>python38Packages.pycparser</li>
    <li>python38Packages.pyflakes</li>
    <li>python38Packages.pygments</li>
    <li>python38Packages.pyparsing</li>
    <li>python38Packages.pysocks</li>
    <li>python38Packages.pytest (python38Packages.pytest_6)</li>
    <li>python38Packages.pytest-asyncio</li>
    <li>python38Packages.pytestcov (python38Packages.pytest-cov)</li>
    <li>python38Packages.pytest-expect</li>
    <li>python38Packages.pytest-forked</li>
    <li>python38Packages.pytest-freezegun</li>
    <li>python38Packages.pytest-mock</li>
    <li>python38Packages.pytestrunner (python38Packages.pytest-runner)</li>
    <li>python38Packages.pytest-subtests</li>
    <li>python38Packages.python_mimeparse</li>
    <li>python38Packages.pytoml</li>
    <li>python38Packages.pytz</li>
    <li>python38Packages.rednose</li>
    <li>python38Packages.ruamel-base (python38Packages.ruamel_base)</li>
    <li>python38Packages.semantic-version</li>
    <li>python38Packages.setuptools</li>
    <li>python38Packages.setuptools-rust</li>
    <li>python38Packages.setuptools-scm (python38Packages.setuptools_scm)</li>
    <li>python38Packages.simplejson</li>
    <li>python38Packages.six</li>
    <li>python38Packages.smartypants</li>
    <li>python38Packages.sortedcontainers</li>
    <li>python38Packages.soupsieve</li>
    <li>python38Packages.sphinxcontrib-applehelp</li>
    <li>python38Packages.sphinxcontrib-devhelp</li>
    <li>python38Packages.sphinxcontrib-htmlhelp</li>
    <li>python38Packages.sphinxcontrib-jsmath</li>
    <li>python38Packages.sphinxcontrib-qthelp</li>
    <li>python38Packages.sphinxcontrib-serializinghtml</li>
    <li>python38Packages.sphinxcontrib-websupport</li>
    <li>python38Packages.sqlparse</li>
    <li>python38Packages.termstyle</li>
    <li>python38Packages.testpath</li>
    <li>python38Packages.text-unidecode</li>
    <li>python38Packages.toml</li>
    <li>python38Packages.tomlkit</li>
    <li>python38Packages.tornado</li>
    <li>python38Packages.traceback2</li>
    <li>python38Packages.typing-extensions</li>
    <li>python38Packages.typogrify</li>
    <li>python38Packages.tzlocal</li>
    <li>python38Packages.u-msgpack-python</li>
    <li>python38Packages.ukpostcodeparser</li>
    <li>python38Packages.unittest2</li>
    <li>python38Packages.urllib3</li>
    <li>python38Packages.wcwidth</li>
    <li>python38Packages.webencodings</li>
    <li>python38Packages.wheel</li>
    <li>python38Packages.whoosh</li>
    <li>python38Packages.wrapt</li>
    <li>python38Packages.zipp</li>
    <li>python38Packages.zope_event</li>
    <li>python38Packages.zope_interface</li>
    <li>python39 (python39Packages.python)</li>
    <li>python39Packages.alabaster</li>
    <li>python39Packages.appdirs</li>
    <li>python39Packages.async-timeout</li>
    <li>python39Packages.atomicwrites</li>
    <li>python39Packages.attrs</li>
    <li>python39Packages.certifi</li>
    <li>python39Packages.colorama</li>
    <li>python39Packages.constantly</li>
    <li>python39Packages.coverage</li>
    <li>python39Packages.dateutil (python39Packages.python-dateutil)</li>
    <li>python39Packages.distlib</li>
    <li>python39Packages.docutils</li>
    <li>python39Packages.eradicate</li>
    <li>python39Packages.filelock</li>
    <li>python39Packages.imagesize</li>
    <li>python39Packages.importlib-metadata</li>
    <li>python39Packages.incremental</li>
    <li>python39Packages.iniconfig</li>
    <li>python39Packages.itsdangerous</li>
    <li>python39Packages.linecache2</li>
    <li>python39Packages.markupsafe</li>
    <li>python39Packages.mock</li>
    <li>python39Packages.more-itertools</li>
    <li>python39Packages.nose</li>
    <li>python39Packages.nosexcover</li>
    <li>python39Packages.olefile</li>
    <li>python39Packages.outcome</li>
    <li>python39Packages.packaging</li>
    <li>python39Packages.parameterized</li>
    <li>python39Packages.pbr</li>
    <li>python39Packages.pexpect</li>
    <li>python39Packages.pip</li>
    <li>python39Packages.pkgconfig</li>
    <li>python39Packages.pluggy</li>
    <li>python39Packages.pretend</li>
    <li>python39Packages.psutil</li>
    <li>python39Packages.ptyprocess</li>
    <li>python39Packages.py</li>
    <li>python39Packages.pyasn1</li>
    <li>python39Packages.pycodestyle</li>
    <li>python39Packages.pycparser</li>
    <li>python39Packages.pyflakes</li>
    <li>python39Packages.pygments</li>
    <li>python39Packages.pyparsing</li>
    <li>python39Packages.pysocks</li>
    <li>python39Packages.pytz</li>
    <li>python39Packages.ruamel-base (python39Packages.ruamel_base)</li>
    <li>python39Packages.semantic-version</li>
    <li>python39Packages.setuptools</li>
    <li>python39Packages.setuptools-rust</li>
    <li>python39Packages.setuptools-scm (python39Packages.setuptools_scm)</li>
    <li>python39Packages.six</li>
    <li>python39Packages.sortedcontainers</li>
    <li>python39Packages.soupsieve</li>
    <li>python39Packages.sphinxcontrib-applehelp</li>
    <li>python39Packages.sphinxcontrib-devhelp</li>
    <li>python39Packages.sphinxcontrib-htmlhelp</li>
    <li>python39Packages.sphinxcontrib-jsmath</li>
    <li>python39Packages.sphinxcontrib-qthelp</li>
    <li>python39Packages.sphinxcontrib-serializinghtml</li>
    <li>python39Packages.sphinxcontrib-websupport</li>
    <li>python39Packages.testpath</li>
    <li>python39Packages.toml</li>
    <li>python39Packages.tornado</li>
    <li>python39Packages.traceback2</li>
    <li>python39Packages.typing-extensions</li>
    <li>python39Packages.tzlocal</li>
    <li>python39Packages.u-msgpack-python</li>
    <li>python39Packages.ukpostcodeparser</li>
    <li>python39Packages.unittest2</li>
    <li>python39Packages.wcwidth</li>
    <li>python39Packages.wheel</li>
    <li>python39Packages.zipp</li>
    <li>python39Packages.zope_event</li>
    <li>subdl</li>
    <li>w3m-batch</li>
    <li>waf</li>
    <li>xhtml1</li>
    <li>xkeyboard_config (xorg.xkeyboardconfig)</li>
    <li>xmlto</li>
    <li>xorg.libICE</li>
    <li>xorg.libX11</li>
    <li>xorg.libXau</li>
    <li>xorg.libXcomposite</li>
    <li>xorg.libXcursor</li>
    <li>xorg.libXdmcp</li>
    <li>xorg.libXext</li>
    <li>xorg.libXfixes</li>
    <li>xorg.libXi</li>
    <li>xorg.libXinerama</li>
    <li>xorg.libXrandr</li>
    <li>xorg.libXrender</li>
    <li>xorg.libXtst</li>
    <li>xorg.libxcb</li>
    <li>xorg.xcbproto</li>
    <li>xorg.xcbutil</li>
    <li>xorg.xcbutilimage</li>
    <li>xorg.xcbutilkeysyms</li>
    <li>xorg.xcbutilrenderutil</li>
    <li>xorg.xcbutilwm</li>
    <li>xorg.xorgproto</li>
  </ul>
</details>

**Comment by lukegb** [comment]
@ofborg build pkgsStatic.openssl

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Comment by vcunat** [comment]
Uh, rebuild of darwin stdenv... that's like a week of time or something.  And the problem happens on master already.

So, what about a simple fix without non-static rebuild to master for now?  For example I now tested:
```diff
--- a/pkgs/development/libraries/openssl/default.nix
+++ b/pkgs/development/libraries/openssl/default.nix
@@ -124,6 +124,7 @@ let
           rm "$out/lib/"*.a
       fi
 
+    '' + ''
       mkdir -p $bin
     '' + lib.optionalString (!stdenv.hostPlatform.isWindows)
       # Fix bin/c_rehash's perl interpreter line
```

**Comment by jonringer** [comment]
I'm fine with this taking another staging cycle, it was just affecting a single python package which I don't use.

**Comment by SuperSandro2000** [comment]
> So, what about a simple fix without non-static rebuild to master for now? For example I now tested:

Lets not fiddle a while to avoid the stdenv rebuild and just push it to staging.

File: pkgs/development/libraries/openssl/default.nix [hunk 1]
```diff
@@ -140,7 +138,8 @@ let
       substituteInPlace $out/bin/c_rehash --replace ${buildPackages.perl}/bin/perl "/usr/bin/env perl"
     '' +
     ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/openssl/default.nix:140]:
```suggestion
    '' + ''
```

**Comment by vcunat** [comment]
Oh, OK.   I didn't realize that it's known only to affect one package.

**Pull request merged**